src/vsr/replica.zig

```diff
@@ -2749,6 +2754,28 @@ pub fn ReplicaType(
             });
             assert(self.state_machine.commit_timestamp < prepare.header.timestamp);
 
+            // Synchronously record this request in our AOF. This can be used for disaster recovery
+            // in the case of catestrophic storage failure. Internally, write() will only return
+            // once the data has been written to disk with O_DIRECT and O_SYNC.
+            //
+            // We run this here, instead of in state_machine, so we can have full access to the VSR
+            // header information. This way we can just log the Prepare in its entirity.
+            //
+            // A minor detail, but this is not a WAL. Hence the name being AOF - since it's similar
+            // to how Redis's Append Only File works. It's also technically possible for a request
+            // to be recorded by the AOF, with the client not having received a response
+            // (eg, a panic right after writing to the AOF before sending the response) but we
+            // consider this harmless due to our requirement for unique IDs.
+            //
+            // It should be impossible for a client to receive a response without the request having
+            // being logged, however.
+            //
+            // TODO: Do we need to record replies...? For safety?
+            // TODO: Is checking for !recovering the right thing to do?
```

Comment:
Only single-replica clusters commit during `status=recovery` anyway, so it probably doesn't matter.

Also, does AOF recovery handle duplicate/backtracking entries ok? e.g. `A, B, C, D, B, C, D`.

Reply:
Another thing to consider: Once we have state transfer, it may be that no single replica has a "complete" AOF; we'll have to take the union of them.

Reply:
I've now handled backtracking / duplicate entries and potentially incomplete or corrupted files with a new subcommand - `aof prepare` - that can take in multiple potentially corrupted and/or incomplete AOFs and output a single one with a consistent hash chain.